### PR TITLE
BUG - Adding fixes for psycopg

### LINKED ISSUE #364: BUG -psycopg error
### Description of the issue

We get import error of psycopg on AWS lambda. I initially thought it was due to `imap-data-access` version`0.10.0`. Initial debugging pointed to that because I tried creating new lambda layer with `imap-data-access` version 0.7.0 through 0.10.0 and deploy the CDK each time with new layer. All versions worked besides version `0.10.0`. Therefore, I thought that was the issue.

Then to test why version `0.10.0` was causing the issue, I created a local working branch. I did not make any change to `imap-data-access` beside updating version to `0.10.1rc2` on it. Then I tried creating package locally using this command `python -m build --sdist` and then copying the zip file to `sds-data-manager/lambda_layer/python` and update `lambda_layer_construct.py` to pip install that independently like this
```
        code_bundle = lambda_.Code.from_asset(
            layer_dependencies_dir,
            bundling=cdk.BundlingOptions(
                image=runtime.bundling_image,
                command=[
                    "bash",
                    "-c",
                    (
                        "pip install -r requirements.txt -t /asset-output/python && "
                        "pip install imap_data_access-0.10.1rc2.tar.gz -t /asset-output/python && "
                        "cp -au . /asset-output/python"
                    ),
                ],
            ),
        )

        super().__init__(
            id=f"{id}-Layer", code=code_bundle, compatible_runtimes=[runtime], **kwargs
        )
```
When I deployed with above changes through CDK, the issue went away on AWS lambda.

But when everything is deployed through `requirements.txt` and through GitHub deploy action, it raises this issue on AWS lambda. I am not sure now what the issue is.

### Steps to reproduce the issue

Deploy CDK to AWS through GitHub Actions or locally from this snapshot of `sds-data-manager`[commit](https://github.com/IMAP-Science-Operations-Center/sds-data-manager/commit/e9a5b698f61fe452b8230835ae97be11aa1a5a2a)

### Expected behavior (What should happen)

No error

### Actual behavior (What does happen)

{
  "errorMessage": "No module named 'psycopg2._psycopg'",
  "errorType": "ModuleNotFoundError",
  "requestId": "150f8d02-8a1c-41c2-bd29-11a9f79778ec",
  "stackTrace": [
    "  File \"/var/task/SDSCode/create_schema.py\", line 29, in lambda_handler\n    Base.metadata.create_all(db.get_engine())\n",
    "  File \"/var/task/SDSCode/database/database.py\", line 27, in get_engine\n    return create_engine(db_uri)\n",
    "  File \"<string>\", line 2, in create_engine\n",
    "  File \"/opt/python/sqlalchemy/util/deprecations.py\", line 281, in warned\n    return fn(*args, **kwargs)  # type: ignore[no-any-return]\n",
    "  File \"/opt/python/sqlalchemy/engine/create.py\", line 599, in create_engine\n    dbapi = dbapi_meth(**dbapi_args)\n",
    "  File \"/opt/python/sqlalchemy/dialects/postgresql/psycopg2.py\", line 690, in import_dbapi\n    import psycopg2\n",
    "  File \"/opt/python/psycopg2/__init__.py\", line 51, in <module>\n    from psycopg2._psycopg import (                     # noqa\n"
  ]
}

### Code Snippet:

```python
Code
```

### Additional notes, affected areas, and suggested fixes

Files and resources that caused this issue could be these:

1. imap-data-access - version 0.10.0
2. mismatch of psycopg = "^3.1" and psycopg2-binary = "^2.9.9" in pyproject.toml file
3. GitHub workflow yaml not installing dependency for psycopg

### PR DESCRIPTION
**Pull request by tech3371**
# Change Summary
closes #364 

## Overview
I don't know for sure what caused the issue. Re-releasing PyPI version fixed the issue.

## Updated Files
- pyproject.toml
- poetry.lock
- lambda_layer/python/requirements.txt
   - updated version of `imap-data-access`
- sds_data_manager/constructs/efs_construct.py
   - fixes lambda handler error
   - removed suffix from event rule

### CODE REVIEW

**Review by maxinelasp — APPROVED** [review]

**Comment by maxinelasp** [comment]
sorry, didn't see this PR, I'm not sure if you still need it but here's an approval!

**Comment by tech3371** [comment]
> sorry, didn't see this PR, I'm not sure if you still need it but here's an approval!

Yep. I may update this PR soon with new version of `imap-data-access` and undo that change.

**Comment by tech3371** [comment]
@maxinelasp I pushed some changes if you can confirm if it looks good still

File: sds_data_manager/constructs/efs_construct.py [hunk 1]
```diff
@@ -184,7 +184,7 @@ def __init__(
             allow_all_outbound=True,
             runtime=aws_lambda.Runtime.PYTHON_3_12,
             code=code,
-            handler="lambda_function.lambda_handler",
+            handler="efs_lambda.lambda_function.lambda_handler",
```
Comment (maxinelasp) [inline sds_data_manager/constructs/efs_construct.py:187]: Why are these changes in here?

Comment (tech3371) [reply sds_data_manager/constructs/efs_construct.py:187]: As I mentioned in Slack, this was another new bug I found. This line change fixes that issue.

**Pull request merged**